src/pip/_internal/cli/cmdoptions.py

```diff
@@ -568,6 +568,25 @@ def prefer_binary():
          'if this option is used.'
 )  # type: Any
 
+use_pep517 = partial(
+    Option,
+    '--use-pep517',
+    dest='use_pep517',
+    action='store_true',
+    default=None,
+    help='Use PEP 517 for building source distributions '
+         '(use --no-use-pep517 to force legacy behaviour).'
+)  # type: Any
+
+no_use_pep517 = partial(
```

Comment:
I believe we do, although `--use-pep517` works somewhat differently from `--no-build-isolation` and `--no-cache-dir` (because both the negative and the positive form of `--use-pep517` mean something - and omitting the flag means something different than either).

We should rationalise the whole boolean option mess, but that's out of scope for this PR IMO. At this point, all I care about is that the PEP517 option works as expected.

Reply:
Yeah. I just don't want us to be adding to that situation.